### PR TITLE
Add new Ubuntu 24.04 CUDA, ROCm, OneAPI machines

### PR DESCRIPTION
**Pull request by stephenswat**
This commit adds three new images based on Ubuntu 24.04, with OneAPI and, optionally, the AMD ROCm and NVIDIA CUDA plugins.

### CODE REVIEW

File: .github/workflows/build-images.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
   push:
     branches:
       - main
+      - "**"
```
Comment (paulgessinger) [inline .github/workflows/build-images.yml:8]:
```suggestion
```

**Comment by paulgessinger** [comment]
Can you reopen this from a branch on the upstream repo? Let me know if you don't have perms.

**Comment by stephenswat** [comment]
> Let me know if you don't have perms.

I do not right now. :(

**Comment by stephenswat** [comment]
Superseded by #117.

**Pull request closed without merging**